libs/web/hooks/src/useProposalVoteForm.ts

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useState } from "react";
+
+import {
+	getVoteAgainstReferendumExtrinsic,
+	getVoteOnProposalExtrinsic,
+	signAndSend,
+} from "@app-gov/service/cennznet";
+import { useCENNZApi, useCENNZWallet } from "@app-gov/web/providers";
+
+export interface ProposalVoteFormState {
+	step: "Idle" | "Sign" | "Submit" | "Complete";
+	status?: "Cancelled" | "Ok" | "NotOk";
+	statusMessage?: string;
+}
+
+export const useProposalVoteForm = (proposalId: number) => {
+	const { api } = useCENNZApi();
+	const { wallet, selectedAccount } = useCENNZWallet();
+	const [formState, setFormState] = useState<ProposalVoteFormState>({
+		step: "Idle",
+	});
+
+	const setFormStep = (step: ProposalVoteFormState["step"]) => {
+		setFormState({ step });
+	};
+
+	const setFormStatus = (
+		status: ProposalVoteFormState["status"],
+		statusMessage?: ProposalVoteFormState["statusMessage"]
+	) => {
+		setFormState((current) => ({ ...current, status, statusMessage }));
+	};
+
+	const onVote = useCallback(
+		async (vote: boolean) => {
+			if (!api || !wallet) return;
+			setFormStep("Sign");
+			try {
+				const extrinsic = getVoteOnProposalExtrinsic(api, proposalId, vote);
+
+				await signAndSend(
+					[extrinsic, selectedAccount?.address, { signer: wallet.signer }],
+					{
+						onHashed() {
+							setFormStep("Submit");
+						},
+
+						onCancelled() {
+							setFormStatus("Cancelled");
+						},
+					}
+				);
+				setFormState({ step: "Complete", status: "Ok" });
+				// eslint-disable-next-line @typescript-eslint/no-explicit-any
+			} catch (error: any) {
+				console.info(error);
+				setFormStatus(
+					"NotOk",
+					`[${error?.code ?? "UNKNOWN"}] ${error?.message}`
+				);
+			}
+		},
+		[api, proposalId, selectedAccount, wallet]
+	);
+	const onVeto = useCallback(async () => {
+		if (!api || !wallet) return;
+		setFormStep("Sign");
+		try {
+			const extrinsic = getVoteAgainstReferendumExtrinsic(api, proposalId);
+			await signAndSend(
+				[extrinsic, selectedAccount?.address, { signer: wallet.signer }],
+				{
+					onHashed() {
+						setFormStep("Submit");
+					},
+
+					onCancelled() {
+						setFormStatus("Cancelled");
+					},
+				}
+			);
+			setFormState({ step: "Complete", status: "Ok" });
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		} catch (error: any) {
+			console.info(error);
+			setFormStatus("NotOk", `[${error?.code ?? "UNKNOWN"}] ${error?.message}`);
+		}
```

Comment:
Might need to run Prettier format

Reply:
hmm nothing changes when I run `yarn format`, how does it show up to you?

Reply:
ahh I got tripped up because the one above it `setFormStatus` not `setFormState`